src/electronic-ids/pcsc/LatEIDIDEMIAv2.cpp

```diff
@@ -29,50 +29,114 @@
 using namespace pcsc_cpp;
 using namespace electronic_id;
 
+struct KeyInfo
+{
+    bool isECC;
+    byte_type id;
+};
+
 struct LatEIDIDEMIAV2::Private
 {
-    std::optional<bool> isUpdated;
+    std::optional<KeyInfo> authKeyInfo;
+    std::optional<KeyInfo> signKeyInfo;
 };
 
+namespace
+{
+constexpr byte_type DEFAULT_AUTH_KEY_ID = 0x81;
+constexpr byte_type DEFAULT_SIGN_KEY_ID = 0x9F;
+
+inline byte_vector readEF_File(const SmartCard& card, const byte_vector& file)
+{
+    auto response = card.transmit({0x00, 0xA4, 0x02, 0x04, file, 0x00});
+    if (!response.isOK()) {
+        THROW(SmartCardError, "Failed to read EF file");
+    }
+    static const byte_vector findLength {0x80, 0x02};
+    auto pos = std::search(response.data.cbegin(), response.data.cend(), findLength.cbegin(),
+                           findLength.cend());
+    if (pos == response.data.cend()) {
+        THROW(SmartCardError, "Failed to read EF file length");
+    }
+    pos += findLength.size();
+    return readBinary(card, size_t(*pos << 8) + *(pos + 1), 0xFF);
+}
+
+inline byte_vector readEF_PrKD(const SmartCard& card)
+{
+    static const byte_vector EF_OD {0x50, 0x31};
+    const auto info = readEF_File(card, EF_OD);
+    static const byte_vector file {0xA0, 0x06, 0x30, 0x04, 0x04, 0x02};
+    auto pos = std::search(info.cbegin(), info.cend(), file.cbegin(), file.cend());
+    if (pos == info.cend()) {
+        THROW(SmartCardError, "EF.PrKD reference not found");
+    }
+    pos += file.size();
+    return readEF_File(card, {*pos, *(pos + 1)});
+}
+
+inline KeyInfo readPrKDInfo(const SmartCard& card, byte_type keyID)
+{
+    const auto data = readEF_PrKD(card);
+    if (data.empty()) {
+        return {false, keyID};
+    }
+    static const byte_vector needle {0x02, 0x02, 0x00};
```

Comment:
`needle` is vague, `keyIDPattern` tells what we are looking for. But let's fix it in the future.